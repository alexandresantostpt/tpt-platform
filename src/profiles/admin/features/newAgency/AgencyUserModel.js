import moment from 'moment'
import React from 'react'
import styled from 'styled-components'

import { i18n } from '@i18n'

import { formats } from '@constants/dateFormats'

import { statusColors } from './constants'
import { isNotNull } from '@/utils/functions'

const StyledStatusIcon = styled.span`
    background-color: ${({ color }) => color};
    border-radius: 100%;
    display: inline-block;
    height: 0.5rem;
    margin-right: 0.2rem;
    width: 0.5rem;
`

class AgencyModel {
    constructor({
        __v,
        _id,
        active,
        agency,
        appTheme,
        blocked,
        businessName,
        cnpj,
        cpfAdm,
        createdAt,
        emailAdm,
        lastAccess,
        name,
        nameAdm,
        masterAgency,
        socialName,
        updatedAt,
        user
    }) {
        this.__v = __v
        this._id = _id
        this._active = active
        this._agency = agency
        this._appTheme = appTheme
        this._blocked = blocked
        this._businessName = businessName
        this._cnpj = cnpj
        this._cpfAdm = cpfAdm
        this._createdAt = moment(createdAt).utc(false)
        this._emailAdm = emailAdm
        this._lastAccess = lastAccess
        this._masterAgency = masterAgency
        this._name = name
        this._nameAdm = nameAdm
        this._socialName = socialName
        this._updatedAt = moment(updatedAt).utc(false)
        this._user = user
    }

    get userV() {
        const v = this.__v || (this.user && this.user.__v)
        return isNotNull(v) ? v : undefined
    }

    get userId() {
        return this._id || (this.user && this.user._id)
    }

    get agencyV() {
        const v = this.agency && this.agency.__v
        return isNotNull(v) ? v : undefined
    }

    get agencyId() {
        return this.agency && this.agency._id
    }

    get agency() {
        return this._agency
    }

    get appTheme() {
        return this._appTheme
    }

    get blocked() {
        return this._blocked || (this.user && this.user.blocked)
    }

    get businessName() {
        return this._businessName
    }

    get cnpj() {
        return this._cnpj
    }

    get cpfAdm() {
        return this._cpfAdm
    }

    get emailAdm() {
        return this._emailAdm
    }

    get lastAccess() {
        return this._lastAccess
    }

    get lastAccessToTable() {
        return moment(this._lastAccess).format(formats.brasilian)
    }

    get masterAgency() {
        return this._masterAgency
    }

    get name() {
        return this._name
    }

    get nameToTable() {
        return this._name || this._user.name
    }

    get nameAdm() {
        return this._nameAdm
    }

    get socialName() {
        return this._socialName || (this.agency && this.agency.socialName)
    }

    get socialNameToTable() {
        if (this.agency) {
            return this.agency.socialName
        }
        return null
    }

    get statusToTable() {
        return (
            <div>
                <StyledStatusIcon color={this.blocked ? statusColors.BLOCKED : statusColors.RELEASED} />
                {i18n.t(`userStatus.${this.blocked ? 'BLOCKED' : 'RELEASED'}`)}
            </div>
        )
    }

    get user() {
        return this._user
    }

    /* eslint-disable complexity */
    toDTO() {
        return {
            agency: {
                __v: this.agencyV,
                _id: this.agencyId || undefined,
                appTheme: this.appTheme || null,
                businessName: this.businessName || null,
                cnpj: this.cnpj || null,
                logoTipo: this.logoTipo || null,
                phoneContact: this.phoneContact || null,
                phoneEmergency: this.phoneEmergency || null,
                socialName: this.socialName || null
            },
            user: {
                __v: this.userV,
                _id: this.userId || undefined,
                blocked: this.blocked.toString() === 'true',
                cpf: this.cpfAdm || null,
                email: this.emailAdm || null,
                name: this.nameAdm || null,
                role: 'ADMIN'
            }
        }
    }

    toForm() {
        return {
            __v: this.__v,
            _id: this.id,
            appTheme: this.appTheme,
            blocked: this.blocked.toString(),
            businessName: this.businessName,
            cnpj: this.agency.cnpj,
            cpfAdm: this.user.cpf,
            emailAdm: this.user.email,
            logoTipo: this.logoTipo,
            nameAdm: this.user.name,
            phoneContact: this.phoneContact,
            phoneEmergency: this.phoneEmergency,
            socialName: this.socialName
        }
    }
}

export default AgencyModel
