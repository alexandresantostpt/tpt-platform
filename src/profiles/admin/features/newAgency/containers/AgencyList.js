import { connect } from 'react-redux'

import {
    requestDelAction,
    requestGetListAction,
    requestGetSingleAction,
    requestResetAction,
    requestSaveDataAction,
    requestSearchAction
} from '../actions'

import { getListSelector, getObjSelector } from '../selectors'

import AgencyList from '../pages/AgencyList'

const mapStateToProps = state => ({
    agencies: getListSelector(state),
    agency: getObjSelector(state)
})

const mapDispatchToProps = {
    requestDelAction,
    requestGetListAction,
    requestGetSingleAction,
    requestResetAction,
    requestSaveDataAction,
    requestSearchAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AgencyList)
