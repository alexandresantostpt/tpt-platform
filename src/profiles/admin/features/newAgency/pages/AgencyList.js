import { events } from '@constants/events'

import PropTypes from 'prop-types'
import PubSub from 'pubsub-js'
import React, { useEffect, useRef, useState } from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

import { i18n } from '@i18n'

import App from '@containers/App'
import Container from '@containers/Container'
import Summary from '@containers/Summary'
import Button from '@components/button/Button'
import FormBuilder from '@components/form/FormBuilder'
import Search from '@components/search/Search'
import Table from '@components/table/Table'
import FormButtonContainer from '@components/form/container/button/FormButtonContainer'

import { getUser } from '@utils/auth'

import { agencyBasicFormFields } from '../fields'

import AgencyUserModel from '../AgencyUserModel'
import CancelForm from '@/components/cancel/CancelForm'
import If from '@/components/if/If'
import Modal from '@components/modal/ConfirmationModal'

import { isNotNull } from '@utils/functions'

const StyledHeader = styled.div`
    display: flex;
    justify-content: space-between;
    margin: 0 1.1875rem 3.5rem;
`
const Text = styled.div`
    color: ${colors.link};
    font-size: 1.063rem;
    font-weight: 300;
    line-height: 1.53;
    text-align: center;
    width: 18.75rem;
`

const AgencyList = props => {
    const { requestResetAction, requestSearchAction, agencies, agency } = props
    const [modalItem, updateModalItem] = useState(null)
    const [showModal, updateShowModal] = useState(false)
    const [showNewAgencyForm, updateShowNewAgencyForm] = useState(false)
    const form = useRef()
    const masterAgency = getUser() && getUser().agency

    useEffect(() => {
        const { requestGetListAction } = props
        requestGetListAction()
        PubSub.subscribe(events.OPEN_MODAL, handleOpenModal)
    }, [])

    const callSaveData = values => {
        const { requestSaveDataAction } = props
        requestSaveDataAction({
            data: new AgencyUserModel({
                ...values,
                agency: agency && agency.agency,
                masterAgency: masterAgency,
                user: agency && agency.user
            }).toDTO()
        })
        reset()
        updateShowNewAgencyForm(false)
    }

    const del = row => {
        const { requestDelAction } = props
        const { userId } = row
        const { agencyId } = row
        if (agencyId) {
            requestDelAction({ agencyId, userId })
        }
        reset()
    }

    const gettBtnText = () => {
        if (agency && agency.agencyId) {
            return i18n.t('buttons.save')
        }
        return i18n.t('buttons.add')
    }

    const getSingle = row => {
        const agencyId = row.agency._id
        const userId = row.userId || row.user._id
        const id = `${agencyId}/${userId}`
        const { requestGetSingleAction } = props
        requestGetSingleAction({ id })
    }

    const onSaveClick = () => form.current.form.submit()

    const reset = () => {
        PubSub.publish(events.FORM_RESET)
        requestResetAction()
    }

    const handleClickNewAgency = () => {
        reset()
        updateShowNewAgencyForm(true)
    }

    const handleClickCancel = () => {
        reset()
        updateShowNewAgencyForm(false)
    }

    const handleCloseModal = () => {
        updateShowModal(false)
    }

    const handleOpenModal = (msg, item) => {
        updateShowModal(true)
        updateModalItem(item)
    }

    if (agency && agency.agencyId) {
        const requiredFields = ['cnpj', 'cpfAdm', 'emailAdm']
        agencyBasicFormFields.forEach(field => (field.readOnly = requiredFields.includes(field.name)))
    } else {
        agencyBasicFormFields.forEach(field => (field.readOnly = false))
    }

    return (
        <App>
            <Container>
                <StyledHeader>
                    <Search onType={requestSearchAction} placeholder={i18n.t('placeholders.searchAgency')} />
                    <Button
                        color={colors.lightBlueGray}
                        disabled={!agency && showNewAgencyForm}
                        onClick={handleClickNewAgency}
                        text={i18n.t('buttons.newAgency')}
                    />
                </StyledHeader>
                <Table
                    columns={['statusToTable', 'socialNameToTable', 'nameToTable', 'lastAccessToTable', 'actions']}
                    data={agencies}
                    onDelete={del}
                    onEdit={getSingle}
                />
            </Container>
            <Summary>
                <If condition={showNewAgencyForm || isNotNull(agency)}>
                    <FormBuilder
                        fields={agencyBasicFormFields}
                        initialValues={agency ? agency.toForm() : {}}
                        onSubmit={callSaveData}
                        ref={process.env.__TEST__ ? null : form}
                    />
                    <FormButtonContainer>
                        <Button color={colors.purple} disabled={false} onClick={onSaveClick} text={gettBtnText()} />
                        <CancelForm action={handleClickCancel} />
                    </FormButtonContainer>
                </If>
            </Summary>
            <Modal close={handleCloseModal} confirmationAction={del} icon={'alert'} item={modalItem} openModal={showModal}>
                <Text>{i18n.t('modal.aboutToDeleteAgency')}</Text>
                <Text>{i18n.t('modal.continue')}</Text>
            </Modal>
        </App>
    )
}

AgencyList.defaultProps = {
    agency: {}
}

AgencyList.propTypes = {
    agencies: PropTypes.arrayOf(AgencyUserModel).isRequired,
    agency: PropTypes.objectOf(AgencyUserModel),
    requestDelAction: PropTypes.func.isRequired,
    requestGetListAction: PropTypes.func.isRequired,
    requestGetSingleAction: PropTypes.func.isRequired,
    requestResetAction: PropTypes.func.isRequired,
    requestSaveDataAction: PropTypes.func.isRequired,
    requestSearchAction: PropTypes.func.isRequired
}

export default AgencyList
