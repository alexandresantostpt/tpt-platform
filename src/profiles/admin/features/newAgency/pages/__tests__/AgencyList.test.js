import configureMockStore from 'redux-mock-store'
import React from 'react'
import renderer from 'react-test-renderer'

import { List, Map, Record } from 'immutable'
import { Provider } from 'react-redux'

import AgencyList from '../AgencyList'

import AgencyUserModel from '../../AgencyUserModel'

describe('Tests for Agency component', () => {
    let mockStore = {}
    let store = {}
    let task = {}
    let toast = {}

    const agency = new AgencyUserModel({
        agency: {
            __v: 0,
            _id: '5caf547ce3f3db3e69103a10',
            cnpj: '12.123.123/1234-12',
            createdAt: '2019-04-11T14:51:40.267Z',
            deleted: false,
            socialName: 'AgÃªncia testes',
            updatedAt: '2019-04-11T19:13:03.592Z'
        },
        user: {
            __v: 0,
            _id: '5caf547ce3f3db3e69103a11',
            active: false,
            agency: '5caf547ce3f3db3e69103a10',
            blocked: true,
            cpf: '437.958.818-11',
            createdAt: '2019-04-11T14:51:40.271Z',
            deleted: false,
            email: 'teste@teste.com.br',
            lastAccess: '2019-04-11T13:47:29.312Z',
            name: 'Adm teste',
            password: 'c705d2857889ded02c1ec111134236e7a4a9cb3d89ec0d3577ee40a84ec180db',
            role: 'ADMIN',
            updatedAt: '2019-04-11T19:13:03.594Z'
        }
    })

    beforeEach(() => {
        task = {
            __v: 28,
            _id: '5c9e31f8c57b4917b426e334',
            createdAt: '2019-03-29T14:55:52.764Z',
            describe: 'teste 123',
            done: false,
            updatedAt: '2019-03-29T14:55:52.764Z',
            user: '5c9dfb65d1cf42bd5fa66b3d'
        }
        toast = { icon: 'check', message: 'success', type: 'success' }

        mockStore = configureMockStore()
        store = mockStore(
            Record({
                tasksReducers: Map({
                    list: List.of(task)
                }),
                toastReducers: Map({
                    list: List.of(toast)
                })
            })()
        )
    })

    it('Should render withou crash', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <AgencyList
                        agencies={[agency]}
                        agency={agency}
                        requestGetListAction={jest.fn()}
                        requestGetSingleAction={jest.fn()}
                        requestSaveDataAction={jest.fn()}
                        requestSearchAction={jest.fn()}
                    />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
