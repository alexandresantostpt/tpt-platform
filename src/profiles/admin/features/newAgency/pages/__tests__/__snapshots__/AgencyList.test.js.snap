// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Agency component Should render withou crash 1`] = `
Array [
  <header
    className="Header__StyledHeader-od5ehu-0 hIVEkQ"
  >
    <span
      className="Brand__StyledBrand-sc-19fy8m0-0 jmSzTQ"
    />
    <nav
      className="Menu__StyledMenu-w0f985-0 kGkgdR"
    >
      <ul
        className="MenuList__StyledMenuList-sc-1bxu43p-0 ixFBrS"
      >
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-plane Icon__StyledIcon-crr1cg-0 dYwZuZ"
            disabled={false}
            fontSize="2.5rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/"
          >
            Viagens
          </a>
        </li>
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-clients Icon__StyledIcon-crr1cg-0 bEFLZZ"
            disabled={false}
            fontSize="2rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/client"
          >
            Clientes
          </a>
        </li>
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-bookmark Icon__StyledIcon-crr1cg-0 jtTFhN"
            disabled={false}
            fontSize="2.3rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/library"
          >
            Biblioteca
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="Header__StyledOptions-od5ehu-1 hmosuQ"
    >
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <img
          alt="Johnny Silva"
          className="DropdownPicture__StyledDropdownPicture-ylb9qy-0 iHpJFt"
          src="https://api.adorable.io/avatars/100/tpt-platform"
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Meu perfil
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Registro de atividades
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Atualizado às 8:45
            </span>
          </li>
        </ul>
      </div>
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="icon icon-notification Icon__StyledIcon-crr1cg-0 llBDkg"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Notificações
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Alteração serviço
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Há 2 minutos
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Lembrete: Preencher informações
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Ontem às 15:30
            </span>
          </li>
        </ul>
      </div>
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="icon icon-config Icon__StyledIcon-crr1cg-0 llBDkg"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Configurações
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={[Function]}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Alterar senha
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 hOZFkb"
            onClick={[Function]}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Sair
            </span>
          </li>
        </ul>
      </div>
    </div>
  </header>,
  <section
    className="App__SideBar-h8qnv1-0 cfDVBS"
  >
    <h1>
      TravelCalendar
    </h1>
    <div
      className="Tasks__StyledTasks-sc-1hy8ads-0 cjVtaH"
    >
      <h2
        className="Title__StyledTitle-sc-11bxhsf-0 frSblw"
      >
        Tarefas
      </h2>
      <button
        className="AddButton__StyledAddButon-ppf0uf-0 gwSZMh"
        onClick={[Function]}
      >
        <span
          className="icon icon-plus Icon__StyledIcon-crr1cg-0 jfRxiR"
          disabled={false}
          fontSize="0.8rem"
          onClick={null}
        />
        Adicionar Tarefa
      </button>
      <div
        className="Task__StyledTask-sc-1fi2258-0 cEDnPQ"
      >
        <span
          className="TaskCheck__StyledTaskCheck-sc-1tgftv-0 kYiGLh"
          onClick={[Function]}
        >
          <button
            className="TaskCheck__StyledButton-sc-1tgftv-1 elQAyk"
          />
          <span
            className="icon icon-check Icon__StyledIcon-crr1cg-0 llBDkg"
            disabled={false}
            fontSize="1.75rem"
            onClick={null}
          />
        </span>
        <div
          className="TaskBody__StyledBody-sc-1p8y2k3-0 cOetdt"
        >
          <textarea
            autoComplete="off"
            className="TaskEdit__StyledTaskEdit-f5yune-0 kXgwgR"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            value="teste 123"
          />
        </div>
        <span
          className="TaskActions__StyledTaskActions-sc-92wkgs-0 fGXkbW"
        >
          <span
            className="icon icon-times Icon__StyledIcon-crr1cg-0 efWoYG"
            disabled={false}
            fontSize="1.5rem"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
  </section>,
  <main
    className="Container__StyledContainer-sc-1i8buc6-0 SsRsD"
  >
    <div
      className="AgencyList__StyledHeader-sc-1kva3no-0 eXFpEK"
    >
      <div
        className="Search__StyledSearch-sc-1rsqafp-0 iZTykq"
      >
        <input
          className="Search__StyledSearchInput-sc-1rsqafp-1 gOzgfR"
          onChange={[Function]}
          placeholder="Pesquise agências"
        />
        <span
          className="icon icon-search Icon__StyledIcon-crr1cg-0 cmTdMB"
          disabled={false}
          fontSize="0.9rem"
          onClick={null}
        />
      </div>
      <button
        className="Button__StyledButton-sc-1fc3klx-0 cyrdMB"
        disabled={false}
        fill={false}
        onClick={[Function]}
      >
        Nova agência
      </button>
    </div>
    <table
      className="Table__StyledTable-y4is3j-0 fabprq"
    >
      <thead>
        <tr
          className="TableHeaderRow__StyledTableHeaderRow-bvb8hb-0 Nmgc"
        >
          <th
            className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 cdraHa"
            onClick={[Function]}
          >
            Status
             
            <span
              className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 llBDkg"
              disabled={false}
              fontSize="1.75rem"
              onClick={null}
            />
          </th>
          <th
            className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 cdraHa"
            onClick={[Function]}
          >
            Agência
             
            <span
              className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 llBDkg"
              disabled={false}
              fontSize="1.75rem"
              onClick={null}
            />
          </th>
          <th
            className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 cdraHa"
            onClick={[Function]}
          >
            Administrador
             
            <span
              className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 llBDkg"
              disabled={false}
              fontSize="1.75rem"
              onClick={null}
            />
          </th>
          <th
            className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 cdraHa"
            onClick={[Function]}
          >
            Último acesso
             
            <span
              className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 llBDkg"
              disabled={false}
              fontSize="1.75rem"
              onClick={null}
            />
          </th>
          <th
            className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 cdraHa"
            onClick={[Function]}
          >
            
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </main>,
  <section
    className="Summary__StyledSummary-sc-1mglm3x-0 fKeRKj"
  >
    <form
      className="FormBuilder__StyledForm-c1htou-0 PZxQa"
      onSubmit={[Function]}
    >
      <div
        className="form__container"
      >
        <div
          className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
        >
          <label
            className="FormLabel__StyledFormLabel-vu10xa-0 dTqfRJ"
          >
            Nome da agência
            
            <span
              className="FormRequired__StyledFormRequired-svb7d8-0 eWTUXS"
            >
               *
            </span>
          </label>
          <input
            className="FormField__Field-sc-1at7ilr-0 bMCrTq FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
            data-cy="input-socialName"
            disabled={false}
            id="socialName"
            maxLength={255}
            name="socialName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder=""
            readOnly={false}
            required={true}
            type="text"
            value="Agência testes"
          />
        </div>
        <div
          className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
        >
          <label
            className="FormLabel__StyledFormLabel-vu10xa-0 dTqfRJ"
          >
            CNPJ
            
            <span
              className="FormRequired__StyledFormRequired-svb7d8-0 eWTUXS"
            >
               *
            </span>
          </label>
          <input
            className="FormField__Field-sc-1at7ilr-0 bMCrTq FormInputMask__StyledFormInputMask-sc-1ltqnwb-0 leVwm"
            data-cy="input-cnpj"
            disabled={false}
            id="cnpj"
            isInvalid={false}
            name="cnpj"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  3,
                  3,
                  4,
                  2,
                ],
                "delimiters": Array [
                  ".",
                  ".",
                  "/",
                  "-",
                ],
              }
            }
            readOnly={true}
            required={true}
            type="mask"
            value="12.123.123/1234-12"
          />
        </div>
        <div
          className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
        >
          <label
            className="FormLabel__StyledFormLabel-vu10xa-0 dTqfRJ"
          >
            Nome do administrador
            
            <span
              className="FormRequired__StyledFormRequired-svb7d8-0 eWTUXS"
            >
               *
            </span>
          </label>
          <input
            className="FormField__Field-sc-1at7ilr-0 bMCrTq FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
            data-cy="input-nameAdm"
            disabled={false}
            id="nameAdm"
            maxLength={255}
            name="nameAdm"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder=""
            readOnly={false}
            required={true}
            type="text"
            value="Adm teste"
          />
        </div>
        <div
          className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
        >
          <label
            className="FormLabel__StyledFormLabel-vu10xa-0 dTqfRJ"
          >
            CPF do administrador
            
            <span
              className="FormRequired__StyledFormRequired-svb7d8-0 eWTUXS"
            >
               *
            </span>
          </label>
          <input
            className="FormField__Field-sc-1at7ilr-0 bMCrTq FormInputMask__StyledFormInputMask-sc-1ltqnwb-0 leVwm"
            data-cy="input-cpfAdm"
            disabled={false}
            id="cpfAdm"
            isInvalid={false}
            name="cpfAdm"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            options={
              Object {
                "blocks": Array [
                  3,
                  3,
                  3,
                  2,
                ],
                "delimiters": Array [
                  ".",
                  ".",
                  "-",
                ],
              }
            }
            readOnly={true}
            required={true}
            type="mask"
            value="437.958.818-11"
          />
        </div>
        <div
          className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
        >
          <label
            className="FormLabel__StyledFormLabel-vu10xa-0 dTqfRJ"
          >
            E-mail do administrador
            
            <span
              className="FormRequired__StyledFormRequired-svb7d8-0 eWTUXS"
            >
               *
            </span>
          </label>
          <input
            className="FormField__Field-sc-1at7ilr-0 bMCrTq FormInput__StyledFormInput-sc-77s1r2-0 jcCQHd"
            data-cy="input-emailAdm"
            disabled={false}
            id="emailAdm"
            maxLength={255}
            name="emailAdm"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder=""
            readOnly={true}
            required={true}
            type="text"
            value="teste@teste.com.br"
          />
        </div>
        <div
          className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
        >
          <select
            className="FormSelect__StyledFormSelect-sc-1iyz1uk-0 CPsvA"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              hidden={true}
              value=""
            >
              Definição de acesso...
            </option>
            <option
              value="false"
            >
              Liberado
            </option>
            <option
              value="true"
            >
              Bloqueado
            </option>
          </select>
        </div>
      </div>
    </form>
    <div
      className="FormButtonContainer-sc-1cqqivn-0 ADyLt"
    >
      <button
        className="Button__StyledButton-sc-1fc3klx-0 cyrdMB"
        disabled={false}
        fill={false}
        onClick={[Function]}
      >
        Salvar
      </button>
      <a
        className="CancelForm__StyledCancel-jf7x9v-0 hTbSZg"
        onClick={[Function]}
      >
        Cancelar
      </a>
    </div>
  </section>,
]
`;
